lib/internal/util/types.js

```diff
@@ -2,10 +2,18 @@
 
 const {
   ArrayBufferIsView,
+  Object,
   ObjectDefineProperties,
   TypedArrayPrototypeGetSymbolToStringTag,
 } = primordials;
 
+function isPlainObject(value) {
+  if (typeof value !== 'object') return false;
+  if (value.constructor && value.constructor !== Object) return false;
```

Comment:
I think this is subject to prototype pollution. What about using `ObjectGetPrototypeOf(value)` (and compare with `ObjectPrototype` and `null`)?

Reply:
That’s essentially https://github.com/sindresorhus/is-plain-obj/blob/main/index.js:

```js
export default function isPlainObject(value) {
	if (Object.prototype.toString.call(value) !== '[object Object]') {
		return false;
	}

	const prototype = Object.getPrototypeOf(value);
	return prototype === null || prototype === Object.prototype;
}
```

Reply:
@targos I think a simple `if (!value) return false` covers where `value` is `null`?

@GeoffreyBooth that looks more convoluted. Is that `'[object Object]'` reliable? Also, wouldn't `prototype === Object.prototype` be subject to prototype pollution also well? If all those are 👍, sure.

Reply:
I’m no expert on these kind of fundamental helpers. My instinct is to borrow from MDN, Lodash, Underscore or @sindresorhus as those are very “vetted” sources where all the edge cases have probably been hammered out by thousands of educated reviewers. If @targos or @aduh95 or someone else has a version they feel confident in, I vote for using what they suggest.

Reply:
> Also, wouldn't prototype === Object.prototype be subject to prototype pollution also well?

There's no risk if you use `const { ObjectPrototype } = primordials`

Reply:
> Or do we have to do this “is plain object” check at all? Could we just check specifically for the properties we’re expecting?

I think that's what we should do indeed, AFAIK there's nowhere in the codebase where we try to enforce objects to be  "plain".

Reply:
I think we can't do that because we don't have specific properties to look for: I believe a user is allowed to void context (by providing an empty plain object), thus making all properties optional. If some properties are required, then nevermind, we can do that.

But the user is not allow to provide an invalid type.

Reply:
I think we should stick with the `validateObject` validator, there's no reason to enforce the passed object to an instance of `Object` is there?
If we want to ensure the object is indeed "plain", we should use the suggestion in https://github.com/nodejs/node/pull/42623#discussion_r847520491 (but using primordials as targos pointed out).